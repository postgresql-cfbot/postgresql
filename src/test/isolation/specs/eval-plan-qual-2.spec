setup
{
 CREATE TABLE accounts (accountid text PRIMARY KEY, balance numeric not null,
   balance2 numeric GENERATED ALWAYS AS (balance * 2) STORED);
 INSERT INTO accounts VALUES ('checking', 600), ('savings', 600);
}

teardown
{
 DROP TABLE accounts;
}

session s1
setup		{ BEGIN ISOLATION LEVEL READ COMMITTED; }
step wb1	{ DELETE FROM accounts WHERE balance = 600 RETURNING *; }
step c1		{ COMMIT; }

session s2
setup		{ BEGIN ISOLATION LEVEL READ COMMITTED; }
step wx2	{ UPDATE accounts SET balance = balance + 450 WHERE accountid = 'checking' RETURNING balance; }
step c2	{ COMMIT; }

session s3
setup		{ BEGIN ISOLATION LEVEL READ COMMITTED; }
step read	{ SELECT * FROM accounts ORDER BY accountid; }
step read_u	{ SELECT * FROM accounts; }

teardown    { COMMIT; }

permutation read_u wx2 wb1 c2 c1 read_u read
