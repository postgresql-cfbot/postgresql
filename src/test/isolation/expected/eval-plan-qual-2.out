Parsed test spec with 3 sessions

starting permutation: read_u wx2 wb1 c2 c1 read_u read
step read_u: SELECT * FROM accounts;
accountid|balance|balance2
---------+-------+--------
checking |    600|    1200
savings  |    600|    1200
(2 rows)

step wx2: UPDATE accounts SET balance = balance + 450 WHERE accountid = 'checking' RETURNING balance;
balance
-------
   1050
(1 row)

step wb1: DELETE FROM accounts WHERE balance = 600 RETURNING *; <waiting ...>
step c2: COMMIT;
step wb1: <... completed>
accountid|balance|balance2
---------+-------+--------
savings  |    600|    1200
(1 row)

step c1: COMMIT;
step read_u: SELECT * FROM accounts;
accountid|balance|balance2
---------+-------+--------
checking |   1050|    2100
(1 row)

step read: SELECT * FROM accounts ORDER BY accountid;
accountid|balance|balance2
---------+-------+--------
checking |   1050|    2100
(1 row)

